src/tests/suite/daos_container.c

```diff
@@ -1596,6 +1596,165 @@ co_modify_acl_access(void **state)
 	test_teardown((void **)&arg);
 }
 
+static void
+expect_ownership(test_arg_t *arg, d_string_t user, d_string_t grp)
+{
+	int			 rc;
+	daos_prop_t		*prop;
+	struct daos_prop_entry	*entry;
+
+	prop = daos_prop_alloc(2);
+	assert_non_null(prop);
+
+	prop->dpp_entries[0].dpe_type = DAOS_PROP_CO_OWNER;
+	prop->dpp_entries[1].dpe_type = DAOS_PROP_CO_OWNER_GROUP;
+
+	rc = daos_cont_query(arg->coh, NULL, prop, NULL);
+	assert_int_equal(rc, 0);
+
+	entry = daos_prop_entry_get(prop, DAOS_PROP_CO_OWNER);
+	assert_non_null(entry);
+	assert_string_equal(entry->dpe_str, user);
+
+	entry = daos_prop_entry_get(prop, DAOS_PROP_CO_OWNER_GROUP);
+	assert_non_null(entry);
+	assert_string_equal(entry->dpe_str, grp);
+
+	daos_prop_free(prop);
+}
+
+static void
+co_set_owner(void **state)
+{
+	test_arg_t	*arg0 = *state;
+	test_arg_t	*arg = NULL;
+	d_string_t	 original_user;
+	d_string_t	 original_grp;
+	d_string_t	 new_user = "newuser@";
+	d_string_t	 new_grp = "newgrp@";
+	int		 rc;
+
+	rc = test_setup((void **)&arg, SETUP_EQ, arg0->multi_rank,
+			DEFAULT_POOL_SIZE, NULL);
+	assert_int_equal(rc, 0);
+
+	while (!rc && arg->setup_state != SETUP_CONT_CONNECT)
+		rc = test_setup_next_step((void **)&arg, NULL, NULL, NULL);
+	assert_int_equal(rc, 0);
```

Comment:
haven't looked at test_setup() carefully right now but is it possible to make a single call specifying SETUP_CONT_CONNECT as the argument, rather than have the test_setup call followed by the next_step loop?

Reply:
I think that is possible in the cases where we don't care about the props, like this one.